### ISSUE TITLE
Revise Worker/Queue management

### ISSUE THREAD

**Issue by drf** [appscode/voyager#1386]
The current worker/queue implementation in Voyager is apparently the cause of most bugs/failures, especially connected to LEGO. Currently, every worker job has a backoff of 1 second and a rate limit of 5. This means Voyager is losing operations (e.g.: a certificate will not renew if there's a temporary failure with the ACME challenge), and most of all it is not capable of handling temporary failures.

My proposal:
* Increase the base delay time to 10 seconds
* Rather than using  DefaultControllerRateLimiter, use ItemExponentialFailureRateLimiter with a base of 10 seconds and an upper limit of 10 minutes
* Increase the rate limit of each operation to a more reasonable number (e.g.: 50)

I can also put a patch together if you agree with the approach.

**Comment by tamalsaha** [comment]
Thanks @drf ! I think we could make this changes for the Certificate queue to handle the issues around ACME/Lego. Do you think we need these changes for the main Ingress part?

**Comment by drf** [comment]
Hey @tamalsaha , I think a good approach would be acting here https://github.com/appscode/voyager/blob/master/vendor/kmodules.xyz/client-go/tools/queue/worker.go#L22 .

We might want to expose those parameters when creating the worker, and let each queue/worker decide on its own what to do. Also, I'm not even sure the module is capable of coping with different mechanisms of backoff - it uses Get() to process messages, disregarding entirely whether the item has a backoff or not. So ultimately I think the fix should go in the client-go module. What do you think?

**Comment by tamalsaha** [comment]
Yes. The change should be made in https://github.com/kmodules/client-go/blob/master/tools/queue/worker.go . That is used in many places. So, we need a way to make the change so that existing behavior can be preserved in other places.

**Comment by drf** [comment]
Personally, I would just make another New that takes all those new arguments, stick the old New to the current behaviour and ensure that Worker handles Queue policies correctly (arguably, currently it totally disregards any directive from the Queue itself and the returned object)

**Comment by tamalsaha** [comment]
Sounds good. Looking forward to your pr.
